perses/rjmc/topology_proposal.py

```diff
@@ -2847,49 +3552,83 @@ def _get_mol_atom_map(current_molecule, proposed_molecule, atom_expr=None, bond_
             _logger.warning('atom_expr not set. using DEFAULT')
             bond_expr = DEFAULT_BOND_EXPRESSION
 
+        if matching_criterion == 'name':
+            #assert the names are unique:
+            for _mol in [current_molecule, proposed_molecule]:
+                if SmallMoleculeSetProposalEngine.check_molecule_name_uniqueness(current_molecule):
```

Comment:
Think this should be `SmallMoleculeSetProposalEngine.check_molecule_name_uniqueness(_mol)` otherwise I don't get what the loop is for?